internal/services/appservice/helpers/windows_web_app_schema.go

```diff
@@ -438,90 +440,195 @@ func SiteConfigSchemaWindowsComputed() *pluginsdk.Schema {
 	}
 }
 
-func ExpandSiteConfigWindows(siteConfig []SiteConfigWindows, existing *web.SiteConfig, metadata sdk.ResourceMetaData, servicePlan web.AppServicePlan) (*web.SiteConfig, *string, error) {
-	if len(siteConfig) == 0 {
-		return nil, nil, nil
-	}
-
+func (s *SiteConfigWindows) ExpandForCreate(appSettings map[string]string) (*web.SiteConfig, error) {
 	expanded := &web.SiteConfig{}
-	if existing != nil {
-		expanded = existing
+
+	expanded.AlwaysOn = pointer.To(s.AlwaysOn)
+	expanded.AcrUseManagedIdentityCreds = pointer.To(s.UseManagedIdentityACR)
+	expanded.AutoHealEnabled = pointer.To(s.AutoHeal)
+	expanded.FtpsState = web.FtpsState(s.FtpsState)
+	expanded.HTTP20Enabled = pointer.To(s.Http2Enabled)
+	expanded.LoadBalancing = web.SiteLoadBalancing(s.LoadBalancing)
+	expanded.LocalMySQLEnabled = pointer.To(s.LocalMysql)
+	expanded.ManagedPipelineMode = web.ManagedPipelineMode(s.ManagedPipelineMode)
+	expanded.MinTLSVersion = web.SupportedTLSVersions(s.MinTlsVersion)
+	expanded.RemoteDebuggingEnabled = pointer.To(s.RemoteDebugging)
+	expanded.ScmIPSecurityRestrictionsUseMain = pointer.To(s.ScmUseMainIpRestriction)
+	expanded.ScmMinTLSVersion = web.SupportedTLSVersions(s.ScmMinTlsVersion)
+	expanded.Use32BitWorkerProcess = pointer.To(s.Use32BitWorker)
+	expanded.WebSocketsEnabled = pointer.To(s.WebSockets)
+	expanded.VirtualApplications = expandVirtualApplications(s.VirtualApplications)
+	expanded.VnetRouteAllEnabled = pointer.To(s.VnetRouteAllEnabled)
+
+	if s.ApiManagementConfigId != "" {
+		expanded.APIManagementConfig = &web.APIManagementConfig{
+			ID: pointer.To(s.ApiManagementConfigId),
+		}
 	}
 
-	winSiteConfig := siteConfig[0]
+	if s.ApiDefinition != "" {
+		expanded.APIDefinition = &web.APIDefinitionInfo{
+			URL: pointer.To(s.ApiDefinition),
+		}
+	}
 
-	currentStack := ""
-	if len(winSiteConfig.ApplicationStack) == 1 {
-		winAppStack := winSiteConfig.ApplicationStack[0]
-		currentStack = winAppStack.CurrentStack
+	if s.AppCommandLine != "" {
+		expanded.AppCommandLine = pointer.To(s.AppCommandLine)
 	}
 
-	if servicePlan.Sku != nil && servicePlan.Sku.Name != nil {
-		if isFreeOrSharedServicePlan(*servicePlan.Sku.Name) {
-			if winSiteConfig.AlwaysOn {
-				return nil, nil, fmt.Errorf("always_on cannot be set to true when using Free, F1, D1 Sku")
+	expanded.AppSettings = ExpandAppSettingsForCreate(appSettings)
+
+	if len(s.ApplicationStack) == 1 {
+		winAppStack := s.ApplicationStack[0]
+		if winAppStack.NetFrameworkVersion != "" {
+			expanded.NetFrameworkVersion = pointer.To(winAppStack.NetFrameworkVersion)
+		}
+		if winAppStack.NetCoreVersion != "" {
+			expanded.NetFrameworkVersion = pointer.To(winAppStack.NetCoreVersion)
```

Comment:
double-checking: Core's still reusing the Framework param?

Reply:
Yes, it's the same property, this is actually only still there for legacy support for existing resources, I don't believe it possible to create this anymore, it will need revisiting at the next major version.